ui/src/dashboards/components/CellEditorOverlay.tsx

```diff
@@ -185,170 +93,95 @@ class CellEditorOverlay extends Component<Props, State> {
-        onResetFocus={this.handleResetFocus}
-      />
-    )
-  }
-
-  private get formattedSources(): SourcesModels.SourceOption[] {
-    const {sources} = this.props
-    return sources.map(s => ({
-      ...s,
-      text: `${s.name} @ ${s.url}`,
-    }))
+    })
   }
 
   private onRef = (r: HTMLDivElement) => {
     this.overlayRef = r
   }
 
-  private queryStateReducer = (
-    queryTransition
-  ): CellEditorOverlayActionsFunc => (queryID: string, ...payload: any[]) => {
-    const {queriesWorkingDraft} = this.state
-    const queryWorkingDraft = queriesWorkingDraft.find(q => q.id === queryID)
-
-    const nextQuery = queryTransition(queryWorkingDraft, ...payload)
-
-    const nextQueries = queriesWorkingDraft.map(q => {
-      if (q.id === queryWorkingDraft.id) {
-        return {...nextQuery, source: nextSource(q, nextQuery)}
-      }
-
-      return q
-    })
-
-    this.setState({queriesWorkingDraft: nextQueries})
-  }
-
-  private handleChangeService = (
-    selectedService: Service,
-    selectedSource: SourcesModels.Source
-  ) => {
-    const queriesWorkingDraft: QueriesModels.QueryConfig[] = this.state.queriesWorkingDraft.map(
-      q => ({
-        ..._.cloneDeep(q),
-        source: selectedSource,
-      })
-    )
-    this.setState({selectedService, selectedSource, queriesWorkingDraft})
-  }
-
-  private handleAddQuery = () => {
-    const {queriesWorkingDraft} = this.state
-    const newIndex = queriesWorkingDraft.length
-
-    this.setState({
-      queriesWorkingDraft: [
-        ...queriesWorkingDraft,
-        {...defaultQueryConfig({id: uuid.v4()}), source: this.initialSource},
-      ],
-    })
-    this.handleSetActiveQueryIndex(newIndex)
-  }
-
-  private handleDeleteQuery = index => {
-    const {queriesWorkingDraft} = this.state
-    const nextQueries = queriesWorkingDraft.filter((__, i) => i !== index)
-
-    this.setState({queriesWorkingDraft: nextQueries})
-  }
-
   private handleSaveCell = () => {
-    const {queriesWorkingDraft, isStaticLegend} = this.state
-    const {cell, thresholdsListColors, gaugeColors, lineColors} = this.props
+    const {isStaticLegend} = this.state
+    const {
+      queryDrafts,
+      thresholdsListColors,
+      gaugeColors,
+      lineColors,
+      cell,
+    } = this.props
 
-    const queries: DashboardsModels.CellQuery[] = queriesWorkingDraft.map(q => {
-      const timeRange = q.range || {
-        upper: null,
-        lower: TEMP_VAR_DASHBOARD_TIME,
-      }
-      const source = getDeep<string | null>(q.source, 'links.self', null)
+    const queries: DashboardsModels.CellQuery[] = queryDrafts.map(q => {
+      const queryConfig = _.get(q, 'queryConfig')
+      const timeRange = getTimeRange(queryConfig)
+      const source = getDeep<string | null>(
+        queryConfig,
+        'source.links.self',
+        null
+      )
       return {
-        queryConfig: q,
-        query: q.rawText || buildQuery(TYPE_QUERY_CONFIG, timeRange, q),
+        ...q,
```

Comment:
yea. so this was written in preparation for flux being apart of all of this as well. instead of just keeping track of query configs, we are keeping track of queries in a cell which each have their query config. So query drafts is of the type CellQuery.

Reply:
👍